bsmetadata/preprocessing_tools/html_parser/objects.py

```diff
@@ -51,8 +51,8 @@ def convert_html_metadata_dataclass_to_dict(metadata: Metadata):
             "relative_end_pos": metadata.relative_end_pos,
             # The information about the HTML tag is separated into two keys because the dictionary must have a stable
             # format between the different types of metadata
-            # the comment have a specific type that is not recognized by pyArrow when we load it in a dataset
-            "value": str(metadata.value.tag) if metadata.value.tag == Comment else metadata.value.tag,
+            # some tags have a value of type `cython_function_or_method` which is not supported by pyrarrow
+            "value": str(metadata.value.tag) if type(metadata.value.tag) == type(Comment) else metadata.value.tag,
```

Comment:
This addition avoids another type of error when working with the datasets library (it does not accept cpython formats)